#WeatherPy Homework....after starter files jumped to pdf format, so flying blind
update: starter file back up after fixing 'merge' problem in gitbash...yay! all but output file established properly without having starter code visible :) 
Currently have api_key setup slightly differently...but looks like its working for me this way


looking to plot
* Temperature (F) vs. Latitude
* Humidity (%) vs. Latitude
* Cloudiness (%) vs. Latitude
* Wind Speed (mph) vs. Latitude
From openweather API will need current weather for locating cities by latitude? or just map layers to collect humidity, cloudiness, and wind speed?

In [1]:
#start with importing dependencies...need pandas, matplotlib, time, citipy, json? better safe than sorry, requests for weather api
import numpy as np
import pandas as pd
import requests
import time
from citipy import citipy
import json
import matplotlib.pyplot as plt



In [2]:
#found my api key...now off to remember how to use it...
from api_keys import api_key
api_key

'772c40118f5ede93501eb573841310af'

notes from Tuesday, 8/7
#setup openweathermaps url for requests
#url = https://openweathermap.org/api  ...this doesn't display url when asked, returning to openweather class assignment for proper use of requests and api entry

# Save config information
url = "http://api.openweathermap.org/data/2.5/weather?"
city = "London"

# Build query URL
query_url = url + "appid=" + api_key + "&q=" + city
query_url


#testing
response = requests.get(url).json()
print(json.dumps(response, indent=4, sort_keys=True))
doesn't work with 'invalid api key' error thrown...

This is the setup that was used in instructor example for weather:
#Get Weather Data
weather_response = requests.get(query_url)
weather_json = weather_response.json()

#get the temperature from the response
print(f"The weather API responded with: {weather_json}.")


#trying with weather example to test api request
# Get weather data
weather_response = requests.get(query_url)
weather_json = weather_response.json()

# Get the temperature from the response
print(f"The weather API responded with: {weather_json}.")

#Same error thrown, removing the "appid=' from api_keys...worked the second time with change

need to review citipy documentation, and exercises on creating library and looping for unique values so I can pull data for at 
least 500 cities...will require creating a dictionary i.e. cities = []

In looking at 'movies' activity, want to try editing above query as such:
change 'city' to cities = ["London", "Saint Louis", "Madrid"]
responses = []

for city in cities
    temp_data = requests.get(query_url + city).json()
    responses.append(temp_data)
tested in 'weatherpy practice', getting errors due to list of cities, not a string
need to get to class now!




In [3]:
#starter code and instructions indicate need for output csv file...need to review process for when/where this is created
output_data_file = "output_data/cities.csv"

In [4]:
#after starter restored, see need to setup lat/long range to gather city data
lat_range = (-90, 90)
lng_range = (-180, 180)


picked up again on Thursday 8/9 after falling asleep with computer late on 8/7 and crazy day on 8/8

In [5]:
#from starter for generating cities list
# List for holding lat_lngs and cities
lat_lngs = []
cities = []

# Create a set of random lat and lng combinations
lats = np.random.uniform(low=-90.000, high=90.000, size=1500)
lngs = np.random.uniform(low=-180.000, high=180.000, size=1500)
lat_lngs = zip(lats, lngs)

# Identify nearest city for each lat, lng combination
for lat_lng in lat_lngs:
    city = citipy.nearest_city(lat_lng[0], lat_lng[1]).city_name
    
    # If the city is unique, then add it to a our cities list
    if city not in cities:
        cities.append(city)

# Print the city count to confirm sufficient count
len(cities)

639

I have sufficient city count, now how to make the api calls with appropriate time delay?

Time limitation on calls....from API documentation, only attempt call once per 10 minutes, free accountinfo says:
Threshold: 7,200 
Hourly forecast: 5 
Daily forecast: 0 
Calls 1min: 60 
If limits exceeded, 1st block=1hr, increases by 1hr until 4 blocks, at 5 blocks increases to 24 hrs. 
With 639 cities, that requires 11 calls, hence 2 hours just to call. I'm screwed and waited too long to do this, as usual, and after assignment was extended due to need to limit requests by time :(

API doc also says calls need to use city_ID. So, downloaded city list.json, unzipped, and saved in folder with working docs. This data only includes id#, name, lat/lng

Where is my "output_data/cities.csv" file? I can't find it in my working directory, but remember we talked about this last week in an exercise with citipy...My assumption is that this is not generated until I have actual data to save from making the calls. 

Next, need to make df to store temp, humid, cloud, wind for plots...is that one df or 4?

In [ ]:
#starter code for calling API
# OpenWeatherMap API Key
# api_key = api_keys.api_key.....how does this change since I imported api key? I think it should work since "api_key" exists

# Starting URL for Weather Map API Call 
url = "http://api.openweathermap.org/data/2.5/weather?units=Imperial&APPID=" + api_key 
